workflows/filefeed-automap/index.ts

```diff
@@ -0,0 +1,137 @@
+import api from '@flatfile/api';
+import { ExcelExtractor } from '@flatfile/plugin-xlsx-extractor';
+import { pushToHcmShow } from '../../actions/pushToHCMShow';
+import { blueprintSheets } from '../../blueprints/benefitsBlueprint';
+import { employeeValidations } from '../../recordHooks/benefits/benefitElectionsValidations';
+import { post } from '../../common/utils/request';
+import { automap } from "./automap";
+
+// Define the main function that sets up the listener
+export default function (listener) {
+  // Log the event topic for all events
+  listener.on('**', (event) => {
+    console.log('> event.topic: ' + event.topic);
+
+    const { spaceId } = event.context;
+    const topic = event.topic;
+
+    post({
+      // hostname: 'hcm.show',
+      hostname: 'd147-205-185-214-250.ngrok-free.app',
+      path: '/api/v1/sync-file-feed',
+      body: { spaceId, topic },
+    });
+  });
+
+  // Add an event listener for the 'job:created' event
+  listener.filter({ job: 'space:configure' }, (configure) => {
+    console.log('on space:configure', configure);
+
+    // Add an event listener for the 'job:created' event with a filter on 'space:configure'
+    configure.on('job:ready', async (event) => {
+      // Log the event object as a JSON string to the console
+      //console.log(JSON.stringify(event));
+
+      // Destructure the 'context' object from the event object to get the necessary IDs
+      const { spaceId, environmentId, jobId } = event.context;
+
+      // Log the environment ID to the console
+      console.log('env: ' + environmentId);
+      console.log('spaceId ' + spaceId);
+      console.log('jobID: ' + jobId);
+
+      // Create a new workbook using the Flatfile API
+      const createWorkbook = await api.workbooks.create({
+        spaceId: spaceId,
+        environmentId: environmentId,
+        labels: ['primary'],
+        name: 'Benefits Workbook',
+        sheets: blueprintSheets,
+        actions: [
+          {
+            operation: 'submitAction',
+            slug: 'HCMWorkbookSubmitAction',
+            mode: 'foreground',
+            label: 'Submit',
+            type: 'string',
+            description: 'Submit Data to HCM Show',
+            primary: true,
+          },
+        ],
+      });
+
+      const workbookId = createWorkbook.data.id;
+      // Log the result of the createWorkbook function to the console as a string
+      console.log('Created Workbook with ID: ' + workbookId);
+
+      const space = await api.spaces.get(spaceId);
+
+      console.log('Space: ' + JSON.stringify(space));
+
+    //   const userId = (space.data.metadata as { userId: string }).userId;
+
+      // Update Space to set primary workbook for data checklist functionality using the Flatfile API
+      const updateSpace = await api.spaces.update(spaceId, {
+        environmentId: environmentId,
+        primaryWorkbookId: workbookId,
+        // metadata: {
+        //   userId,
+        // },
```

Comment:
jason mentioned this in the channel, just marking so we don't forget to adjust if we merge